### PR TITLE
Update Searching to Liferay 7.1

### PR DESCRIPTION
**Pull request by lmarqs**
Check the [README.md](https://github.com/lmarqs/liferay-training/blob/update-searching/README.md)

### CODE REVIEW

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelDocumentContributor.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        Locale defaultLocale = null;
+
+        try {
+            defaultLocale = PortalUtil.getSiteDefaultLocale(entry.getGroupId());
+        } catch (PortalException e) {
+            throw new Exception(e);
+        }
+
+        document.addDate(Field.MODIFIED_DATE, entry.getModifiedDate());
+        document.addText("email", entry.getEmail());
```
Comment (lmarqs) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelDocumentContributor.java]: Why doesn't searching using email work?

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookKeywordQueryContributor.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        immediate = true,
+        property = "indexer.class.name=br.com.objective.training.model.Guestbook",
+        service = KeywordQueryContributor.class
+)
+public class GuestbookKeywordQueryContributor implements KeywordQueryContributor {
+
+    @Override
+    public void contribute(String keywords, BooleanQuery booleanQuery, KeywordQueryContributorHelper keywordQueryContributorHelper) {
+        /**
+         * @lmarqs: No localized term is declared on GuestbookSearchRegistrar#activate.
```
Comment (lmarqs) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookKeywordQueryContributor.java]: Can we simply delete this class?

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelDocumentContributor.java [hunk 3]
```diff
@@ -46,37 +48,39 @@ public void contribute(Document document, Entry entry) {
...
+            Guestbook guestbook = _guestbookLocalService.getGuestbook(guestbookId);
 
-        String guestbookName = guestbook.getName();
+            String guestbookName = guestbook.getName();
 
-        String localizedGbName = LocalizationUtil.getLocalizedName("guestbookName", defaultLocale.toString());
+            String localizedGbName = LocalizationUtil.getLocalizedName("guestbookName", defaultLocale.toString());
+
+            document.addText(localizedGbName, guestbookName);
+        } catch (PortalException pe) {
```
Comment (lmarqs) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelDocumentContributor.java]: Should we throw this exception?

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java [hunk 4]
```diff
@@ -42,19 +44,20 @@ public void contribute(Document document, Guestbook guestbook) {
...
-        Locale defaultLocale = null;
+            Locale defaultLocale = defaultLocale = PortalUtil.getSiteDefaultLocale(guestbook.getGroupId());
+            String localizedField = LocalizationUtil.getLocalizedName(Field.TITLE, defaultLocale.toString());
 
-        try {
-            defaultLocale = PortalUtil.getSiteDefaultLocale(guestbook.getGroupId());
-        } catch (PortalException e) {
-            throw new Exception(e);
+            document.addText(localizedField, guestbook.getName());
+        } catch (PortalException pe) {
```
Comment (lmarqs) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java]: Should we throw this exception?

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelSummaryContributor.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * <p>
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.calendar.internal.search;
```
Comment (ifnazar) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelSummaryContributor.java]: Please, fix the package

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package br.com.objective.training.search;
+
+import br.com.objective.training.model.Guestbook;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.search.Document;
+import com.liferay.portal.kernel.search.Field;
```
Comment (ifnazar) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java]: Please, remove unused import

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java [hunk 7]
```diff
@@ -42,19 +44,20 @@ public void contribute(Document document, Guestbook guestbook) {
         /**
          * @lmarqs: The code below has been copied from GuestbookIndexer#doGetDocument.
          */
+        try {
+            document.addDate(Field.MODIFIED_DATE, guestbook.getModifiedDate());
 
-        document.addDate(Field.MODIFIED_DATE, guestbook.getModifiedDate());
-
-        Locale defaultLocale = null;
+            Locale defaultLocale = defaultLocale = PortalUtil.getSiteDefaultLocale(guestbook.getGroupId());
```
Comment (ifnazar) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java]: Please, remove the duplicate attribution  "defaultLocale = defaultLocale ="

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelSummaryContributor.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * <p>
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.calendar.internal.search;
```
Comment (ifnazar) [inline modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelSummaryContributor.java]: Please, fix the package

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelSummaryContributor.java (hunk 5 above)
Comment (lmarqs) [reply modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/EntryModelSummaryContributor.java]: Fixed

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java (hunk 6 above)
Comment (lmarqs) [reply modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java]: Fixed

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java (hunk 7 above)
Comment (lmarqs) [reply modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelDocumentContributor.java]: Fixed

File: modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelSummaryContributor.java (hunk 8 above)
Comment (lmarqs) [reply modules/guestbook/guestbook-service/src/main/java/br/com/objective/training/search/GuestbookModelSummaryContributor.java]: Fixed

**Review by ifnazar — APPROVED** [review]

**Pull request merged**